src/plugins/intel_gpu/src/graph/include/primitive_inst.h

```diff
@@ -260,6 +261,9 @@ class primitive_inst {
 
     std::vector<memory::cptr> _intermediates_memory;
 
+    static const size_t _weights_cache_capacity = 2;
+    LruCache<size_t, memory::ptr> _reordered_weights_cache;
```

Comment:
I think we need to share weights across streams somehow to reduce memory consumption

Reply:
Yeah, agree, do you mind to implement this in separate PR?